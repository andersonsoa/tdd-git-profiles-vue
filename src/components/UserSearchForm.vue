<template>
  <div class="user-search__container">
    <input
      v-model="username"
      @keydown.enter="pesquisar"
      class="user-search__input"
      name="username"
      type="text"
    />
    <button
      @click="pesquisar"
      class="user-search__button"
      type="submit"
      name="submit"
    >
      Pesquisar
    </button>
  </div>
</template>

<script>
export default {
  name: "UserSearchForm",
  data: () => ({
    username: ""
  }),
  methods: {
    pesquisar() {
      this.$emit("submitted", this.username);
      this.username = "";
    }
  }
};
</script>

<style scoped>
.user-search__container {
  width: 100%;
  height: 100px;
  display: flex;
  align-items: center;
  justify-content: center;
}
.user-search__input {
  padding: 10px 15px;
  font-size: 20px;
  border: none;
  border-radius: 4px;
  box-shadow: 1px 1px 3px #0009;
}
.user-search__input:focus {
  outline: none;
}
.user-search__button {
  margin-left: 10px;
  padding: 10px 15px;
  height: 43px;
}
.user-search__button:focus {
  outline: none;
}
</style>
