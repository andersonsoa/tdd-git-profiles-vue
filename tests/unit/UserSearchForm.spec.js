import { shallowMount } from "@vue/test-utils";
import UserSearchForm from "@/components/UserSearchForm";

describe("UserSearchForm.vue", () => {
  const build = () => {
    const wrapper = shallowMount(UserSearchForm);

    return {
      wrapper,
      input: () => wrapper.find("input[name='username']"),
      button: () => wrapper.find("button[name='submit']")
    };
  };

  // testes
  it("rederiza o componente corretamente", () => {
    const { wrapper } = build();

    // asserções
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renderiza os principais filhos", () => {
    const { input, button } = build();

    expect(input().exists()).toBe(true);
    expect(button().exists()).toBe(true);
  });

  it("invoca o evento 'submitted' quando submeter o form", () => {
    const { wrapper, input, button } = build();
    const expectedUsername = "andersonsoa";

    input().setValue(expectedUsername);

    input().trigger("input");
    button().trigger("click");
    button().trigger("submit");

    expect(wrapper.emitted().submitted[0]).toEqual([expectedUsername]);
  });
});
