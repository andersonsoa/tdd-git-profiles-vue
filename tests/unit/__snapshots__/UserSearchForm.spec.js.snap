// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSearchForm.vue rederiza o componente corretamente 1`] = `
<div class="user-search__container"><input name="username" type="text" class="user-search__input"> <button type="submit" name="submit" class="user-search__button">
    Pesquisar
  </button></div>
`;
